lesson_3/phone_book.py

```diff
@@ -0,0 +1,161 @@
+# Реализовать интерактивное меню с двумя опциями выбора:
+# а) добавление в справочник
+# б) вывод номера телефона по имени
+# в) вывод имени по номеру телефона
+#
+# Пример
+# Add to dict
+# Get telephone
+# Get name
+# 	1
+# 	John, +380953336667
+# 	1
+# 	Alan, +36674490
+# 	2
+# 	Alan
+# 	+36674490
+# 3
+# +380953336667
+# John
+# Посчитать количество номеров телефонов в предыдущем примере, в которых есть 3 или более одинаковых цифр подряд
+#
+#
+# Добавить возможность удаления человека из справочника
+# Добавить возможность редактирования номера\имени
+import re
+
+
+def get_number(question_string='? :'):
+    h = ''
+    while not h.isnumeric():
+        h = input(question_string)
+        if h.isnumeric():
+            return int(h)
+        elif h == '':
+            return 0
+        print('is not numeric \nplease put a number')
+
+
+def get_phone(question_string='Phone :'):
+    h = ''
+    while not h.isnumeric() or h[:1] == '+' and h[1:].isnumeric():
+        h = input(question_string)
+        if h.isnumeric() or h[:1] == '+' and h[1:].isnumeric():
+            return h
+        print('please put a correct phone')
+
+
+def add_contact(contact_name, contact_phone, phonebook):
+    h = {contact_name: contact_phone}
+    phonebook.update(h)
+    return phonebook.get(contact_name)
+
+
+def get_contact_phone(contact_name, phonebook):
+    h = phonebook.get(contact_name)
+    if h:
+        return h
+    else:
+        return None
+
+
+def get_contact_name(contact_phone, phonebook):
+    for k, v in phonebook.items():
+        if v == contact_phone:
+            return k
+    return 'This phone not found'
+
+
+def remove_contact(contact_name, phonebook):
+    if not phonebook.get(contact_name):
+        return "This name not found. \nCONTACT NOT"
+    h = phonebook.pop(contact_name)
+    return h
+
+
+def edit_contact(contact_name, new_phone, phonebook):
+    if not phonebook.get(contact_name):
+        return "This name not found. \nCONTACT NOT"
```

Comment:
Never return error as string. You can use raise Exception()
```suggestion
        raise Exception('Name not found')
```